### PR TITLE
lua: Move to lua 5.4 and include by default v4

### PR DESCRIPTION
**Pull request by J0eJ0h**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4776

Describe changes:

- Move to using lua 5.4
- Remove luajit support and --enable-luajit configure option
- Change --enable-lua option to --disable-lua and build with lua by default

https://github.com/OISF/suricata/pull/10238 with:
- Fixes to PR only CI workflows

### CODE REVIEW

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -1518,14 +1518,15 @@ jobs:
                 libyaml-dev \
                 libcap-ng-dev \
                 libcap-ng0 \
+                liblua5.4-dev \
```
Comment (jlucovsky) [inline .github/workflows/builds.yml:1521]: This package is listed twice.

File: .github/workflows/builds.yml [hunk 2]
```diff
@@ -2419,7 +2425,6 @@ jobs:
                 libmagic-dev \
                 libjansson-dev \
                 libgeoip-dev \
-                liblua5.1-dev \
```
Comment (jlucovsky) [inline .github/workflows/builds.yml:2422]: lua5.4 not needed?

**Review by jlucovsky — COMMENTED** [review]
Should lua jit be entirely removed?

`git grep luajit` returns results from the `src` directory.

File: .github/workflows/builds.yml (hunk 1 above)
Comment (J0eJ0h) [reply .github/workflows/builds.yml:1521]: Oops.  Doesn't break anything but easy to fix

File: .github/workflows/builds.yml (hunk 2 above)
Comment (J0eJ0h) [reply .github/workflows/builds.yml:2422]: lua 5.4 is not packaged for Debian 10

**Comment by J0eJ0h** [comment]
> Should lua jit be entirely removed?
> 
> `git grep luajit` returns results from the `src` directory.

The remaining instances should just be references to it for rules and flowvars, not the luajit library\code itself.

**Comment by J0eJ0h** [comment]
Comments addressed in https://github.com/OISF/suricata/pull/10263

**Pull request closed without merging**